### PR TITLE
ydict: init at 2.2.0

### PR DESCRIPTION
**Pull request by zendo**
###### Description of changes

ydict is a comand-line  dictionary for Chinese.

[https://github.com/TimothyYe/ydict](url)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/ydict/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    repo = "ydict";
+    rev = "refs/tags/v${version}";
+    sha256 = "sha256-zhjsXZsRk0UNijjqjGjZh4TiPxAM5p+ic3JMx2wrPeY=";
+  };
+
+  vendorSha256 = "sha256-O6czDfKD18rGVMIZv6II09oQu1w0ijQRuZRGt2gj9Cs=";
+
+  ldflags = [ "-s" "-w" "-X=main.Version=${version}" ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ydict/default.nix:18]:
```suggestion
  # has no tests
  doCheck = false;
```

File: pkgs/applications/misc/ydict/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ lib, fetchFromGitHub, buildGoModule }:
+
+buildGoModule rec {
+  pname = "ydict";
+  version = "2.2.0";
+
+  src = fetchFromGitHub {
+    owner = "TimothyYe";
+    repo = "ydict";
+    rev = "refs/tags/v${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ydict/default.nix:10]:
```suggestion
    rev = "v${version}";
```
not required

File: pkgs/applications/misc/ydict/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  # has no tests
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A command-line Chinese dictionary";
+    homepage = "https://github.com/TimothyYe/ydict";
+    license = licenses.mit;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ zendo ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ydict/default.nix:26]: Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME`` into this PR.

Comment (zendo) [reply pkgs/applications/misc/ydict/default.nix:26]: Thank for your help.

**Pull request merged**